lib/rules/no-useless-escape.js

```diff
@@ -23,6 +23,12 @@ const VALID_STRING_ESCAPES = [
     "\r"
 ];
 
+const VALID_TEMPLATE_ESCAPES = VALID_STRING_ESCAPES.concat([
+    "$",
+    "{",
```

Comment:
Minor nit: I don't think this will report something like `` `foo\{bar}` `` or `` `foo\$bar` ``, even though the escape is unnecessary in both cases. Is that intended?